Library/Homebrew/caveats.rb

```diff
@@ -46,6 +47,18 @@ def zsh_completion_caveats
     end
   end
 
+  def has_fish
+      return system("fish -v > /dev/null 2> /dev/null")
+  end
+
+  def fish_completion_caveats
+    if keg and keg.completion_installed? :fish and has_fish then <<-EOS.undent
```

Comment:
I think `which("fish")` would be enough. And you can remove the `has_fish` method.